File: Backend/app/models/project.py
from tortoise import fields, models
import uuid

class Project(models.Model):
    id = fields.UUIDField(pk=True, default=uuid.uuid4)
    name = fields.CharField(max_length=255)
    description = fields.TextField(null=True)

    org = fields.ForeignKeyField(
        'models.Organization',
        related_name='projects',
        on_delete=fields.CASCADE
    )

    created_by = fields.ForeignKeyField(
        'models.User',
        related_name='created_projects',
        on_delete=fields.SET_NULL,
        null=True
    )

    is_archieved = fields.BooleanField(default=False)

    createdAt = fields.DatetimeField(auto_now_add=True)
    updatedAt = fields.DatetimeField(auto_now=True)

    class Meta:
        table = "projects"
        indexes = [
            ("org",),
            ("org", "is_archieved"),
        ]

File: frontend/src/types/project.ts
import { User } from "@/stores/slices/auth.slice"
import { Organization } from "./organization"

export interface Project {
    id: string
    name: string
    description: string | null
    organization: Organization | null
    createdBy: User | null
    is_archieved: boolean
    createdAt: string
    updatedAt: string
}

export interface CreateProjectInput {
    name: string
    description: string | null
}

export interface UpdateProjectInput {
    name: string | null
    description: string | null
    is_archieved: boolean | null
}